src/tests/unit/onbase-dao.test.js

```diff
@@ -0,0 +1,243 @@
+import 'axios';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+import 'config';
+import setCookie from 'set-cookie-parser';
+import { createConfigStub } from './test-helpers';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('Test emails-dao', () => {
+  const docType1 = {
+    id: '24680',
+    name: 'TestDocType',
+    typeId: '924680',
+  };
+  const fakeData = {
+    data: {
+      access_token: 'fake token',
+      keywordGuid: 'fake guid',
+      keywordCollection: 'fake keyword collection',
+      id: 'fake id',
+      items: [docType1],
+    },
+  };
+
+  const keyword1 = {
+    name: 'Keyword1',
+    typeId: '924680',
+    keywordTypeId: '924680',
+  };
+  const keyword2 = {
+    name: 'Keyword2',
+    typeId: '9123456',
+    keywordTypeId: '9123456',
+  };
+  const keywordType1 = {
+    id: '924680',
+    name: 'KeywordType1',
+  };
+  const keywordType2 = {
+    id: '9123456',
+    name: 'KeywordType2',
+  };
+  const fakeKeywordsTypesData = {
+    data: {
+      access_token: 'fake token',
+      keywordGuid: 'fake guid',
+      keywordCollection: 'fake keyword collection',
+      id: 'fake id',
+      items: [keywordType1, keywordType2],
+    },
+  };
+
+  let stubAxios;
+  let onbaseDao;
+  let stubAxios2;
+  let onbaseDao2;
+
+  beforeEach(() => {
+    createConfigStub();
+    stubAxios = sinon.stub().callsFake(
+      () => fakeData,
+    );
+    onbaseDao = proxyquire('db/http/onbase-dao', {
+      axios: stubAxios,
+    });
+    stubAxios2 = sinon.stub().callsFake(
+      () => fakeKeywordsTypesData,
+    );
+    onbaseDao2 = proxyquire('db/http/onbase-dao', {
+      axios: stubAxios2,
+    });
+    sinon.stub(setCookie, 'parse').returns([{
+      name: 'FB_LB',
+      value: 'fake cookie',
+      path: '/',
+      httpOnly: true,
+      secure: true,
+      sameSite: 'none',
+    }]);
+  });
+  afterEach(() => sinon.restore());
+
+  const testResultWithCookie = (result, data, stub) => {
+    result.should
+      .eventually.be.fulfilled
+      .and.deep.equals([data, 'fake cookie'])
+      .and.to.have.length(2);
+    sinon.assert.calledOnce(stub);
+  };
+  const testResultCookieOnly = (result, stub) => {
+    result.should
+      .eventually.be.fulfilled
+      .and.deep.equals(['fake cookie'])
+      .and.to.have.length(1);
+    sinon.assert.calledOnce(stub);
+  };
+  const testResultData = (result, stub) => {
+    result.should
+      .eventually.be.fulfilled
+      .and.deep.equals(fakeData.data);
+    sinon.assert.calledOnce(stub);
+  };
+  const testResultError = (result, msg, stub) => {
+    result.should
+      .eventually.be.fulfilled
+      .and.have.property('message', msg);
+    sinon.assert.calledOnce(stub);
+  };
+
+  it('getAccessToken for valid profile should return a valid token',
```

Comment:
What if `getAccessToken` got an invalid profile? and how about if the response is not 200?

Same comments with all of your other test cases. Besides successful/fulfilled cases, please also ensure they can catch the error as well.